src/hotspot/share/cds/metaspaceShared.cpp

```diff
@@ -1352,6 +1359,51 @@ MapArchiveResult MetaspaceShared::map_archives(FileMapInfo* static_mapinfo, File
   return result;
 }
 
+#ifdef _LP64
+
+// Protection zone handling.
+//
+// A Klass structure must never be located at the encoding base since that would encode to an
+// invalid nKlass of zero. At runtime, we set the encoding base to the start of the mapped
+// archive. In order to catch accidental accesses via a zero nKlass, we will establis a no-access
+// zone there, similar to how the heap does it. Space for that page must be prepared when dumping
+// the archive.
+static constexpr uint64_t protzone_tag       = 0x50524F545A4F4E45ULL; // "PROTZONE"
+static constexpr uint64_t protzone_tag_start = 0x2D3E2D3E50524F54ULL; // "->->PROT"
+static constexpr uint64_t protzone_tag_end   = 0x50524F543C2D3C2DULL; // "PROT<-<-"
```

Comment:
Is this endianness dependent?

Reply:
I don't think so. We only read archives created on the same architecture. The only thing that could change would be the page size.